geode-core/src/main/java/org/apache/geode/internal/cache/PartitionedRegionDataStore.java

```diff
@@ -494,6 +494,18 @@ CreateBucketResult grabFreeBucket(final int possiblyFreeBucketId, final Distribu
 
       return result;
 
+    } catch (RuntimeException validationException) {
+      // GEODE-3055
+      PartitionedRegion leader = ColocationHelper.getLeaderRegion(this.partitionedRegion);
+      boolean isLeader = leader.equals(this.partitionedRegion);
```

Comment:
What if there is more than one colocated child region and one of the child buckets is successfully created, but the other is not. This code will remove the parent bucket. Will the one that was successfully created also get removed?